### PR TITLE
[SPARK-18776][SS] Make Offset for FileStreamSource corrected formatted in json

### PR DESCRIPTION
**Pull request by tdas**
## What changes were proposed in this pull request?

- Changed FileStreamSource to use new FileStreamSourceOffset rather than LongOffset. The field is named as `logOffset` to make it more clear that this is a offset in the file stream log.
- Fixed bug in FileStreamSourceLog, the field endId in the FileStreamSourceLog.get(startId, endId) was not being used at all. No test caught it earlier. Only my updated tests caught it.

Other minor changes
- Dont use batchId in the FileStreamSource, as calling it batch id is extremely miss leading. With multiple sources, it may happen that a new batch has no new data from a file source. So offset of FileStreamSource != batchId after that batch.

## How was this patch tested?

Updated unit test.

### CODE REVIEW

**Comment by tdas** [comment]
@marmbrus @zsxwing

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSourceOffset.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+package org.apache.spark.sql.execution.streaming
+
+import org.json4s.NoTypeHints
+import org.json4s.jackson.Serialization
+
+/**
+ * Offset for the [[FileStreamSource]].
+ * @param logOffset  Position in the [[FileStreamSourceLog]]
+ */
+case class FileStreamSourceOffset(logOffset: Long) extends Offset {
```
Comment (tdas) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSourceOffset.scala]: I named it logoffset to make a more clear that this is a offset in the FileStreamSourceLog.

**Comment by zsxwing** [comment]
LGTM. Merging to master and 2.1.

**Comment by zsxwing** [comment]
Oh, please also support to read the old format in case RC2 is passed.

**Comment by zsxwing** [comment]
LGTM pending tests

**Comment by tdas** [comment]
Merging to master and 2.1

**Pull request closed without merging**